File: Pocket.Domain/Entities/Transaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Pocket.Domain.Entities;

[Table("Transactions")]
public class Transaction
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid UserId { get; set; }
    public decimal Amount { get; set; }
    public int CurrencyId { get; set; }
    public int TransactionGroupId { get; set; }

    [StringLength(250)]
    public string Description { get; set; } = string.Empty;
    public DateTime TransactionDate { get; set; } = DateTime.Now;

#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
    [ForeignKey("UserId")]
    public User User { get; set; }

    [ForeignKey("CurrencyId")]
    public Currency Currency { get; set; } 

    [ForeignKey("TransactionGroupId")]
    public TransactionGroup TransactionGroup { get; set; }
#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
}

File: Pocket.ClientApp/src/app/core/models/transaction.ts
import { v4 as uuidv4 } from 'uuid';

export class Transaction {
  id: string;
  amount: number;
  currency: string;
  transactionGroup: string;
  description: string;
  transactionDate: Date;

  constructor() {
    this.id = uuidv4();
    this.amount = 0;
    this.currency = '';
    this.transactionGroup = '';
    this.description = '';
    this.transactionDate = new Date();
  }
}

export class TransactionGroup {
  id?: number;
  name: string;

  constructor() {
    this.name = '';
  }
}
